chainstate/tx-verifier/src/transaction_verifier/input_output_policy/constraints_accumulator.rs

```diff
@@ -0,0 +1,519 @@
+                    {
+                        TxOutput::Transfer(value, _)
+                        | TxOutput::LockThenTransfer(value, _, _)
+                        | TxOutput::Burn(value) => {
+                            if let Some(coins) = value.coin_amount() {
+                                self.unconstrained_value = (self.unconstrained_value + coins)
+                                    .ok_or(IOPolicyError::AmountOverflow)?;
+                            }
+                        }
+                        TxOutput::DelegateStaking(coins, _) => {
+                            self.unconstrained_value = (self.unconstrained_value + *coins)
+                                .ok_or(IOPolicyError::AmountOverflow)?;
+                        }
+                        TxOutput::CreateDelegationId(..) => { /* do nothing */ }
+                        TxOutput::CreateStakePool(pool_id, _)
+                        | TxOutput::ProduceBlockFromStake(_, pool_id) => {
+                            let block_distance = chain_config
+                                .as_ref()
+                                .decommission_pool_maturity_distance(block_height);
+                            let pledged_amount = pledge_amount_getter(*pool_id)?
+                                .ok_or(IOPolicyError::PledgeAmountNotFound(*pool_id))?;
+                            match self.timelock_constrained.entry(block_distance) {
+                                Entry::Vacant(e) => {
+                                    e.insert(pledged_amount);
+                                }
+                                Entry::Occupied(mut e) => {
+                                    let new_balance = (*e.get() + pledged_amount)
+                                        .ok_or(IOPolicyError::AmountOverflow)?;
+                                    *e.get_mut() = new_balance;
+                                }
+                            };
+                        }
+                    };
+                }
+                TxInput::Account(account) => {
+                    match account.account() {
+                        AccountSpending::Delegation(_, spend_amount) => {
+                            let block_distance =
+                                chain_config.as_ref().spend_share_maturity_distance(block_height);
+                            match self.timelock_constrained.entry(block_distance) {
+                                Entry::Vacant(e) => {
+                                    e.insert(*spend_amount);
+                                }
+                                Entry::Occupied(mut e) => {
+                                    let new_balance = (*e.get() + *spend_amount)
+                                        .ok_or(IOPolicyError::AmountOverflow)?;
+                                    *e.get_mut() = new_balance;
+                                }
+                            };
```

Comment:
I think this could be simplified with something like:
```rust
    let mut balance = self.timelock_constrained.entry(distance).or_insert(0);
    *balance = (*balance + *amount).ok_or(todo!())?;
```

(one more occurrence of this pattern above)